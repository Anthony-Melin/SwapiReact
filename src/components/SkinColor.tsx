import LabelValueFactory from './__components__/LabelValueFactory.tsx'

export default LabelValueFactory('skincolor', 'Skin color')
