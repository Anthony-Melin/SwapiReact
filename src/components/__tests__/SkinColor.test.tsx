import { describe, it, expect, beforeEach } from 'vitest'
import { render, screen } from '@testing-library/react'
import SkinColor from '../SkinColor.tsx'

describe('SkinColor', () => {
    beforeEach(() => {
        render(<SkinColor value="VALUE" />)
    })

    it('should match snapshot', () => {
        expect(screen.getByTestId('skincolor')).toMatchSnapshot()
    })

    it('should have text', () => {
        expect(screen.getByTestId('skincolor')).toHaveTextContent(
            'Skin color: VALUE'
        )
    })
})
