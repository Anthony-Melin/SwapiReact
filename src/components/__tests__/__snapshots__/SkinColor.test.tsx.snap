// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SkinColor > should match snapshot 1`] = `
<p
  data-testid="skincolor"
>
  Skin color
  : 
  VALUE
</p>
`;
